model/project.go

```diff
@@ -322,22 +323,28 @@ func (c *PluginCommandConf) UnmarshalYAML(unmarshal func(interface{}) error) err
 // we maintain Params for backwards compatibility, but we read from YAML when available, as the
 // given params could be corrupted from the roundtrip
 func (c *PluginCommandConf) unmarshalParams() error {
-	if c.ParamsYAML != "" {
+	if c.ParamsYAML != "" && !isEmptyYAML(c.ParamsYAML) {
 		out := map[string]interface{}{}
 		if err := yaml.Unmarshal([]byte(c.ParamsYAML), &out); err != nil {
 			return errors.Wrapf(err, "error unmarshalling params from yaml")
 		}
 		c.Params = out
 		return nil
 	}
-	bytes, err := yaml.Marshal(c.Params)
-	if err != nil {
-		return errors.Wrap(err, "error marshalling params into yaml")
+	if len(c.Params) != 0 {
```

Comment:
Given that `c.Params` must contain at least one item here, I don't understand how `c.ParamsYAML` could ever be anything other than the empty string. Do you know what sets it to `|\n{}`?

Reply:
....really good catch. I had been trying a different approach with !isEmptyYAML before and didn't factor that in when I added this. This if statement is the only change we need.